spec/controllers/host_controller_spec.rb

```diff
@@ -353,6 +354,48 @@
         expect(response.status).to eq(200)
       end
     end
+
+    context 'called with search text' do
+      it 'render GTL with and saves search_text in the session' do
+        expect_any_instance_of(GtlHelper).to receive(:render_gtl).with match_gtl_options(
+          :model_name                     => 'Host',
+          :parent_id                      => nil,
+          :report_data_additional_options => {
+            :lastaction => 'show_list',
+          },
+        )
+        get :show_list, :params => {'search[text]' => 'foobar'}
+        expect(session.fetch_path(:sandboxes, 'host', :search_text)).to eq('foobar')
+        expect(response.status).to eq(200)
+      end
+    end
+  end
+
+  describe '#report_data' do
+    before(:each) do
+      stub_user(:features => :all)
+      EvmSpecHelper.create_guid_miq_server_zone
+    end
+
+    context 'called with search text' do
+      it 'returns hosts filtered by the search text' do
+        h1
+        h2
+
+        session[:sandboxes] = {}
+        session.store_path(:sandboxes, 'host', :search_text, 'foobar')
+        report_data_request(
+          :model                          => 'Host',
+          :parent_id                      => nil,
+          :report_data_additional_options => {
+            :lastaction => 'show_list',
+          }
+        )
+        results = assert_report_data_response
+        expect(results['data']['rows'].length).to eq(1)
+        expect(results['data']['rows'][0]['long_id']).to eq(h1.id)
```

Comment:
This will need to `eq(h1.id.to_s)` - https://github.com/ManageIQ/manageiq-ui-classic/pull/2902